heat/core/operations.py

```diff
@@ -509,6 +515,494 @@ def sum(x, axis=None, out=None):
+           Default: None -> var of all data calculated
+           NOTE -> if multidemensional var is implemented in pytorch, this can be an iterable. Only thing which muse be changed is the raise
+    bessel : Bool
+             Default: True
+             use the bessel correction when calculating the varaince/std
+             toggle between unbiased and biased calculation of the std
+
+    Returns
+    -------
+    ht.tensor containing the var/s, if split, then split in the same direction as x.
+    """
+    if not isinstance(bessel, bool):
+        raise TypeError("bessel must be a boolean, currently is {}".format(type(bessel)))
+
+    def reduce_vars_elementwise(output_shape_i):
+        """
+        Function to combine the calculated vars together.
+        This does an element-wise update of the calculated vars to merge them together using the merge_vars function.
+        This function operates using x from the var function paramters
+
+        Parameters
+        ----------
+        output_shape_i : iterable
+                         iterable with the dimensions of the output of the var function
+
+        Returns
+        -------
+        ht.tensor of the calculated vars
+        """
+
+        if x.lshape[x.split] != 0:
+            mu = __local_operation(torch.mean, x, out=None, dim=axis)
+            var = __local_operation(torch.var, x, out=None, dim=axis, unbiased=bessel)
+        else:
+            mu = tensor.zeros(output_shape_i)
+            var = tensor.zeros(output_shape_i)
+
+        mu_to_combine = tensor.zeros((x.comm.Get_size(),) + tuple(output_shape_i))
+        mu_tot = tensor.zeros((x.comm.Get_size(),) + tuple(output_shape_i))
+        var_to_combine = tensor.zeros((x.comm.Get_size(),) + tuple(output_shape_i))
+        var_tot = tensor.zeros((x.comm.Get_size(),) + tuple(output_shape_i))
+
+        n_for_merge = tensor.zeros(x.comm.Get_size())
+        n2 = tensor.zeros(x.comm.Get_size())
+        mu_to_combine[x.comm.Get_rank()] = mu
+        var_to_combine[x.comm.Get_rank()] = var
+        n2[x.comm.Get_rank()] = x.lshape[x.split]
+        x.comm.Allreduce(mu_to_combine, mu_tot, MPI.SUM)
+        x.comm.Allreduce(var_to_combine, var_tot, MPI.SUM)
+        x.comm.Allreduce(n2, n_for_merge, MPI.SUM)
+
+        sz = x.comm.Get_size()
+        rem1 = rem2 = 0
+        while True:
+            if sz % 2 != 0:
+                if rem1 != 0 and rem2 == 0:
+                    rem2 = sz - 1
+                elif rem1 == 0:
+                    rem1 = sz - 1
+            splt = int(sz / 2)
+            for i in range(splt):  # todo: multithread for GPU
+                mu_reshape = __local_operation(torch.reshape, mu_tot[i], out=None, shape=(1, int(mu_tot[i].lnumel)))[0]
+                var_reshape = __local_operation(torch.reshape, var_tot[i], out=None, shape=(1, int(mu_tot[i].lnumel)))[0]
+                for en, (mu1, var1, mu2, var2) in enumerate(zip(mu_reshape, var_reshape,
```

Comment:
Same as above for the mean() implementation. Is it possible to avoid the code duplication here?

Reply:
fixed for next commit